src/validate.ts

```diff
@@ -36,18 +36,33 @@ function responseValidation(apiResponse: object, schema: object) {
 export function validate(req: object, apiResponse: object) {
   const responseType = req['inputs'][0]['intent'];
   if (responseType === 'action.devices.SYNC') {
+    //validate traits with attributes schema   
     return responseValidation(apiResponse, SYNC_RESPONSE_SCHEMA);
   } else if (responseType === 'action.devices.QUERY') {
+    //validate with states schema   
     return responseValidation(apiResponse, QUERY_RESPONSE_SCHEMA);
   } else if (responseType === 'action.devices.EXECUTE') {
-    const commands_arr = apiResponse['payload']['commands'];
-    const commandsLength= commands_arr.length;
-    //currently registering as an undefined api response. Needs fixes. 
-    for (let i = 0; i <= commandsLength; i++) {
-      const isOn = commands_arr[i]['states']['on'];
-      const isOnline = commands_arr[i]['states']['online'];
-      return [responseValidation(isOn, ON_OFF_STATES_SCHEMA), responseValidation(isOnline, ON_OFF_STATES_SCHEMA)];
+     //validate with states schema   
+    const commandsStatesExpect = {
```

Comment:
You can move this to a module-level constant with `TITLE_CASE` as it won't change.